### PR TITLE
Image variants

### PR DESCRIPTION
**Pull request by rsumilang**

### CODE REVIEW

File: src/Exception/AssetTypeNotSupported.php [hunk 1]
```diff
@@ -6,14 +6,14 @@
 use Gdbots\Pbj\Exception\HasEndUserMessage;
 use Gdbots\Schemas\Pbjx\Enum\Code;
 
-final class AssetTypeNotSupported extends \RuntimeException implements TrinitiDamException, HasEndUserMessage
+final class AssetTypeNotSupported extends \InvalidArgumentException implements TrinitiDamException, HasEndUserMessage
```
Comment (gdbrown) [inline src/Exception/AssetTypeNotSupported.php]: @rsumilang this should extend the InvalidArgumentException in this namespace

File: src/GetUploadUrlsRequestHandler.php [hunk 2]
```diff
@@ -84,13 +98,17 @@ protected function handle(GetUploadUrlsRequest $request): GetUploadUrlsResponse
                  * todo: Add an errors map that contains the filename hash as the key and any
                  * specifics about the error that we want to reveal to the client.
                  */
+                if (get_class($e) === AssetTypeNotSupported::class) {
```
Comment (gdbrown) [inline src/GetUploadUrlsRequestHandler.php]: @rsumilang this i don't understand.  how is this different from the others that aren't throwing an exception?

Comment (rsumilang) [reply src/GetUploadUrlsRequestHandler.php]: @gdbrown I think the original implementation had an idea that some files could error and some could be successful in which case a map was returned to let you know what was good and bad. I'll remove for now.

**Comment by rsumilang** [comment]
@gdbrown Anymore changes here?

**Comment by gdbrown** [comment]
@rsumilang modified to restore original try/catch which was intentional.  front end deals with missing asset_ids/urls rather than failing entire set.  hence the todo to add messaging eventually to the response about files that might have failed.  new exceptions are left as you had them, assuming your code deals with the exceptions on the front end.

**Pull request merged**